changebot/blueprints/milestone_checker.py

```diff
@@ -0,0 +1,14 @@
+from flask import current_app
+
+from .pull_request_checker import pull_request_check
+
+
+@pull_request_check
+def process_milestone(pr_handler, repo_handler):
+    """
+    A very simple set a failing status if the milestone is not set.
+    """
+    if not pr_handler.milestone:
+            pr_handler.set_status('failure', 'This pull request does not have a milestone set.', current_app.bot_username + ": milestone")
```

Comment:
The missing milestone already triggers a failure, it's currently part of the changelog checker here: https://github.com/astropy/astropy-bot/blob/master/changebot/blueprints/changelog_helpers.py#L76

Reply:
Yes, this is not enabled by default. It's if you are not using the astropy style changelog checker. It's included here as a simple example and test.

Reply:
Ohh, this actually made sense to factor out the milestone from the chaneglog. I just mentioned that we check for it in case you haven't seen it.

And maybe inherit the error message from there as it's good it points out setting the milestone is not something the contributor can fix.